packages/graphql/src/services/graphql.service.ts

```diff
@@ -492,25 +502,35 @@ export class GraphqlService implements BackendService {
   }
 
   /**
-   * Update the pagination component with it's new page number and size
+   * Update the pagination component with it's new page number and size.
    * @param newPage
    * @param pageSize
+   * @param cursorArgs Should be supplied when using cursor based pagination
    */
-  updatePagination(newPage: number, pageSize: number) {
+  updatePagination(newPage: number, pageSize: number, cursorArgs?: PaginationCursorChangedArgs) {
     this._currentPagination = {
       pageNumber: newPage,
       pageSize
     };
 
-    let paginationOptions;
-    if (this.options && this.options.isWithCursor) {
-      paginationOptions = {
-        first: pageSize
-      };
+    let paginationOptions: GraphqlPaginationOption | GraphqlCursorPaginationOption = {};
+    if (this.options?.isWithCursor) {
+      // use cursor based pagination
+      // when using cursor pagination, expect to be given a PaginationCursorChangedArgs as arguments,
+      // but still handle the case where it's not (can happen when initial configuration not pre-configured (automatically corrects itself next updatePageInfo() call))
+      if (cursorArgs && cursorArgs instanceof Object) {
+        // remove pageSize and newPage from cursorArgs, otherwise they get put on the query input string
+        // eslint-disable-next-line
```

Comment:
so what's the reason you need to disable eslint from this line? If it's a simple eslint rule missing for deconstructing into variable, we could modify the `.eslintrc` config and add that rule, depending on what linting you're trying to bypass

Reply:
![Screenshot 2023-10-30 at 11 22 15 am](https://github.com/ghiscoding/slickgrid-universal/assets/143153492/865902a7-55d2-40ee-a1d9-cf4ec53f6036)

I've made the disable rule more specific.

Destructing to exempt `pageSize` and `newPage` was the cleanest way to write this, otherwise I'd be iterating through the object props.